### PR TITLE
Add support for Alt-Left/Right and Alt-Backspace editing

### PR DESCRIPTION
**Pull request by samuela**
Fix https://github.com/fadeevab/cliclack/issues/5.

This PR is built off of https://github.com/fadeevab/cliclack/pull/8, so it would be easiest to merge that one first. To review this diff, it may be easiest to look at the changes from https://github.com/fadeevab/cliclack/pull/9/commits/7e9b848caf594defd58d7288a4feebfdf39f8e23 only.

### CODE REVIEW

File: src/prompt/cursor.rs [hunk 1]
```diff
@@ -34,6 +55,24 @@ impl StringCursor {
         }
     }
 
+    pub fn move_word_left(&mut self) {
+        if self.cursor > 0 {
+            let jumps = word_jump_indices(&self.value);
+            let ix = jumps.binary_search(&self.cursor).unwrap_or_else(|i| i);
+            self.cursor = jumps[std::cmp::max(ix - 1, 0)];
```
Comment (fadeevab) [inline src/prompt/cursor.rs:62]: `ix - 1` will panic if `ix == 0`.
```suggestion
            self.cursor = jumps[ix.saturating_sub(1)];
```

File: src/prompt/cursor.rs [hunk 2]
```diff
@@ -34,6 +55,24 @@ impl StringCursor {
...
+        }
+    }
+
+    pub fn move_word_right(&mut self) {
+        if self.cursor < self.value.len() {
+            let jumps = word_jump_indices(&self.value);
+            let ix = jumps
+                .binary_search(&self.cursor)
+                .map_or_else(|i| i, |i| i + 1);
+            self.cursor = jumps[std::cmp::min(ix, jumps.len() - 1)];
```
Comment (fadeevab) [inline src/prompt/cursor.rs:72]: Well, `jump.len()` will never be zero inside of this `if` condition, however, `saturating_sub` is just safer.
```suggestion
            self.cursor = jumps[std::cmp::min(ix, jumps.len().saturating_sub(1))];
```

File: src/prompt/cursor.rs [hunk 3]
```diff
@@ -34,6 +55,24 @@ impl StringCursor {
         }
     }
 
+    pub fn move_word_left(&mut self) {
```
Comment (fadeevab) [inline src/prompt/cursor.rs:58]: A bit misleading name, I thought it actually moves the words. Maybe this `move_left_by_word` should sound less ambiguous.

File: src/prompt/interaction.rs [hunk 4]
```diff
@@ -86,50 +92,56 @@ pub trait PromptInteraction<T> {
...
+                            Key::Del => cursor.delete_right(),
+                            Key::ArrowLeft => cursor.move_left(),
+                            Key::ArrowRight => cursor.move_right(),
+                            Key::Home => cursor.move_home(),
+                            Key::End => cursor.move_end(),
+
+                            // Alt-Backspace
+                            Key::Char('\u{17}') if word_editing => cursor.delete_word_left(),
+
+                            // Alt-ArrowLeft
```
Comment (fadeevab) [inline src/prompt/interaction.rs:119]: Basically, `Ctrl-Left`/`Ctrl-Right` behave similarly in Bash, could `Ctrl` be enabled as well? Thanks!

**Review by fadeevab — CHANGES_REQUESTED** [review]
Thanks for the PR! Super appreciate! There are a few changes requested!

**Comment by samuela** [comment]
Thanks for the review @fadeevab -- taking a look now! 

I also noticed that while alt+{left, right, backspace} work locally, they don't seem to work over SSH. I'll debug...

**Comment by fadeevab** [comment]
@samuela what are you going to do with this? Does it work as expected? Need to rebase it also.

File: src/prompt/cursor.rs (hunk 1 above)
Comment (samuela) [reply src/prompt/cursor.rs:62]: oh good catch! i'm now realizing that ix will always be >= 1 which is why i never got a panic here. This is due to self.cursor > 0 in the if condition. I'll figure out a way to clean this up

Comment (samuela) [reply src/prompt/cursor.rs]: done in https://github.com/fadeevab/cliclack/pull/9/commits/ae9b394a0abdf057888b3e3ab0a84dd488f20318

File: src/prompt/cursor.rs (hunk 3 above)
Comment (samuela) [reply src/prompt/cursor.rs]: done in https://github.com/fadeevab/cliclack/pull/9/commits/ae9b394a0abdf057888b3e3ab0a84dd488f20318

File: src/prompt/interaction.rs (hunk 4 above)
Comment (samuela) [reply src/prompt/interaction.rs:119]: yes... but i propose doing so in a separate PR :)

**Comment by samuela** [comment]
> @samuela what are you going to do with this? Does it work as expected? Need to rebase it also.

Yup, rebase done and comments addressed. I just spent a few hours tracking down the SSH issues. Turns out that its not really an SSH issue, it's an iTerm configuration issue. So I think this should be ready for merging now.

**Pull request merged**